src/explore-education-statistics-admin/src/modules/find-statistics/components/KeyStatistics.tsx

```diff
@@ -0,0 +1,191 @@
+import { EditableContentBlock } from '@admin/services/publicationService';
+import { releaseContentService } from '@admin/services/release/edit-release/content/service';
+import Button from '@common/components/Button';
+import styles from '@common/modules/find-statistics/components/SummaryRenderer.module.scss';
+import { EditingContext } from '@common/modules/find-statistics/util/wrapEditableComponent';
+import {
+  AbstractRelease,
+  Publication,
+} from '@common/services/publicationService';
+import React, { useContext, useEffect, useState } from 'react';
+import EditableKeyStatTile from './EditableKeyStatTile';
+import KeyIndicatorSelectForm from './KeyIndicatorSelectForm';
+
+export interface KeyStatisticsProps {
+  release: AbstractRelease<EditableContentBlock, Publication>;
+  setRelease: (
+    newRelease: AbstractRelease<EditableContentBlock, Publication>,
+  ) => void;
+  isEditing?: boolean;
+}
+
+const KeyStatistics = ({
+  release,
+  setRelease,
+  isEditing,
+}: KeyStatisticsProps) => {
+  const [keyStats, setKeyStats] = useState<
+    | AbstractRelease<EditableContentBlock, Publication>['keyStatisticsSection']
+    | undefined
+  >(undefined);
+
+  useEffect(() => {
+    if (release.keyStatisticsSection) {
+      setKeyStats(release.keyStatisticsSection);
+    } else {
+      setKeyStats(undefined);
+    }
+  }, [release]);
+
+  if (!keyStats) return null;
+  return (
+    <>
+      {isEditing && (
+        <AddKeyStatistics release={release} setRelease={setRelease} />
+      )}
+      <div className={styles.keyStatsContainer}>
+        {keyStats.content &&
+          keyStats.content.map(stat => {
+            if (stat.dataBlockRequest !== undefined) {
+              return stat.type === 'DataBlock' && stat.dataBlockRequest ? (
+                // @ts-ignore
+                <EditableKeyStatTile
+                  key={stat.id}
+                  {...stat}
+                  isEditing={isEditing}
+                  onRemove={() => {
+                    releaseContentService
+                      .deleteContentSectionBlock(
+                        release.id,
+                        release.keyStatisticsSection.id || '',
+                        stat.id,
+                      )
+                      .then(() => {
+                        setRelease({
+                          ...release,
+                          keyStatisticsSection: {
+                            ...release.keyStatisticsSection,
+                            content:
+                              release.keyStatisticsSection.content &&
+                              release.keyStatisticsSection.content.filter(
+                                contentBlock => contentBlock.id !== stat.id,
+                              ),
+                          },
+                        });
+                      });
+                  }}
+                  onSubmit={values => {
+                    return new Promise(() =>
```

Comment:
Do you need a new Promise here?  Is the service not already returning a Promise?